R/model.R

```diff
@@ -369,22 +374,146 @@ compile_method <- function(quiet = TRUE,
+  }
+  if (is.null(stanc_options[["name"]])) {
+    stanc_options[["name"]] <- model_name
+  }
+  if(make_or_stanc=='make'){
+    char_for_stanc_built_options <- "'"
+  }else{
+    char_for_stanc_built_options <- ""
+  }
+
+  stanc_built_options = c()
+  for (i in seq_len(length(stanc_options))) {
+    option_name <- names(stanc_options)[i]
+    if (isTRUE(as.logical(stanc_options[[i]]))) {
+      stanc_built_options = c(stanc_built_options, paste0("--", option_name))
+    } else {
+      stanc_built_options = c(stanc_built_options, paste0("--", option_name, "=",
+        char_for_stanc_built_options, stanc_options[[i]], char_for_stanc_built_options))
+    }
+  }
+
+  if (cmdstan_version() < "2.24") {
+    prepare_precompiled(cpp_options, quiet)
+  }
+
+  if(make_or_stanc=='make'){
+    stancflags_val <- paste0("STANCFLAGS += ", stancflags_val, paste0(stanc_built_options, collapse = " "))
+    these_args = c(
+      cpp_options_to_compile_flags(cpp_options),
+      stancflags_val
+    )
+  }else{
+    stancflags_val <- trimws(c(stancflags_val, c(stanc_built_options)))
+    these_args = c(stancflags_val)
+  }
+  return(these_args)
+}
+CmdStanModel$set("public", name = "generate_arg_strings", value = generate_arg_strings_method)
+
+run_autoformatter_method <- function(quiet = FALSE,
+                                include_paths = NULL,
+                                stanc_options = list()) {
+
+
+  temp_hpp_file <- tempfile(pattern = "model-", fileext = ".hpp")
+  stanc_options[["o"]] <- temp_hpp_file
+
+  arg_strings = self$generate_arg_strings(quiet=quiet,make_or_stanc='stanc',include_paths=include_paths,
+    stanc_options=stanc_options)
+
+  #given bug in stanc (https://github.com/stan-dev/cmdstan/issues/944),
+  # if pedantic-mode warnings have been requested, first run without --auto-format & with echo=!quiet
+  if('--warn-pedantic' %in% arg_strings){
+    pedantic_run_log <- processx::run(
+      command = stanc_cmd(),
+      args = c(self$stan_file(), arg_strings),
+      wd = cmdstan_path(),
+      echo_cmd = FALSE,
+      echo = !quiet,
+      spinner = quiet && interactive(),
+      stderr_line_callback = function(x,p) {
+        message(x)
+      },
+      error_on_status = FALSE
+    )
+    if (pedantic_run_log$status != 0) {
+      stop("Syntax error found! See the message above for more information.",
+           call. = FALSE)
+    }
+  }
+
+  run_log <- processx::run(
+    command = stanc_cmd(),
+    args = c(self$stan_file(), arg_strings, "--auto-format"),
+    wd = cmdstan_path(),
+    echo_cmd = FALSE,
+    echo = FALSE,
+    spinner = quiet && interactive(),
+    stderr_line_callback = function(x,p) {
+      message(x)
+    },
+    error_on_status = FALSE
+  )
+  if (run_log$status != 0) {
+    stop("Syntax error found! See the message above for more information.",
+         call. = FALSE)
+  }
+  return(run_log)
+}
+CmdStanModel$set("public", name = "run_autoformatter", value = run_autoformatter_method)
```

Comment:
Does this method need to be public or will it only ever be used internally?

Reply:
Ditto with the other one; totally should be private.